cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeleteScheduleStage.java

```diff
@@ -69,7 +70,9 @@ public void process(ApplicationWithPrograms input) throws Exception {
       scheduler.deleteSchedule(input.getApplicationId().program(programType,
                                                                 scheduleSpec.getProgram().getProgramName()),
                                scheduleSpec.getProgram().getProgramType(), scheduleSpec.getSchedule().getName());
-      programScheduler.deleteSchedule(input.getApplicationId().schedule(scheduleSpec.getSchedule().getName()));
```

Comment:
right, I'll add this in `ApplicationRegistrationStage`, that's where schedules with deleted programs are removed